### DISCUSSION TITLE
According to the PICkit 5 Docs, Dx/Ex parts support SPI ISP...

### DISCUSSION THREAD

**Discussion by MX682X** [SpenceKonde/DxCore#462] [General]
Has anyone ever *tried* to see if that works? It would be intresting if this is true...

https://ww1.microchip.com/downloads/aemDocuments/documents/DEV/ProductDocuments/UserGuides/MPLAB-PICkit-5-In-Circuit-Debugger-Users-Guide-50003525.pdf (page 30)

![image](https://github.com/SpenceKonde/DxCore/assets/58419867/23ff9c1d-97f1-4fcc-98a4-966af028374e)

**Comment by SpenceKonde** [comment]
If anyone makes this work I'll eat my hat. (US ideom: that claim is
laughable and has no chance of being true, so I can confidently offer to do
something ridiculous like that if it were, because i am just that certain
its not).

I wouldn't trust that claim. Its ludacris and directly contradictory to the
datasheet and every other piece of information regarding these devices.
Furthermore, even if we knew something like that existed, we couldn't
implement it without an aggressive spy campaign using a working debugging
setup, where it was known to work (since i dont think it is actually a
thing, i dont expect any volunteers). But to do anything with this you
would need significant spying on the communications to understand how the
new features not present on classic avrs were implemented.

I'll ask my guy on the inside about it and get my usual reply of chirping
crickets.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by MX682X** [comment]
I'd say it exists. Datasheets's a bit too new. The documents are probably written by different departments, and somone forgot about a memo.

However, we have to figure out more then just the protocol. We would also have to figure out which pins are used for it.
I thought it would be easier then that. 

I suspect they  decided to add this feature for backward compatibility (Arduino) but then found some problems with it and decided to remove it from the documents, after all UPDI is more intresting. But, as making new layout is expensive in money and time, it still sits somewhere on the die. Maybe they couldn't tear it out without significant problems or it's too smal to matter.

**Reply by SpenceKonde** [reply]
Possible - certainly far from unprecedented (ref: MULFACT on Dx can go to 4k for the PLL, internal oscillator has a 32MHz setting on Dx, and so on). But I also would not be the least bit surprised if that was a screwup in the docs. 

Betcha if it does exist, it uses the default pins of SPI0, except you use reset like SS - just like classic AVRs.
